docs/docs/reference/production-deployment.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Production Deployment
+description: >-
+  This article covers production deployment requirements and concerns
+---
+
+# Production Deployment
+
+This page covers the topic of running Pomerium in a production configuration.  See the [quick start section](../quick-start/) for canned example configurations.  
+
+Please also see [architecture](../#architecture) for information on component interactions.
+
+## Service Mode
+
+### All in One
+
+In smaller deployments or while testing, it may be desirable to run in "all-in-one" mode.  This reduces resource footprint and simplifies DNS configuration.  All URLs point at the same pomerium service instance.
+
+### Dedicated Services
+
+In larger footprints, it is recommended that you run Pomerium as a collection of independent services.  This limits blast radius in the event of vulnerabilities and allows for independing [scaling](#scaling) and monitoring of services.
+
+In more complicated deployments, you may also choose to use a single Authorize and/or Authenticate cluster to service multiple proxy clusters.
```

Comment:
Should we give an example of where this makes sense? I'm curious myself! Do you have the MSP / multi-client type deployment in mind?

Reply:
Yes I think two "special" cases exist.  Both probably work framed as MSP type workloads.  

(1) You have a central AuthN and AuthZ but need to support a number of logical proxy groups.  There's no hard boundary.

(2) You have a central control plane but different clusters (customers?) need different IDPs so you might have distributed AuthN and Proxy but central AuthZ

I suppose this really boils down to explaining what creates administrative boundaries between the services.  I think I can flesh this out more.

Reply:
Makes complete sense. I'd list those as examples even, they are helpful.

Reply:
Added a bit more here.  Does my memory serve me correctly on the cookie/domain requirement?